compiler/bindings/python/iree/compiler/tools/import_onnx/__main__.py

```diff
@@ -32,25 +31,44 @@ def main(args: argparse.Namespace):
 
     imp: Any = None
     if args.externalize_params:
-        imp = IREENodeImporter.define_function(
-            model_info.main_graph, m, args.num_elements_threshold, args.params_scope
+        is_stdout = (args.output_file is None) or (args.output_file == "-")
+        output_dir = (
+            Path(args.input_file).parent if is_stdout else Path(args.output_file).parent
+        )
+        output_stem = "imported_model" if is_stdout else Path(args.output_file).stem
+        default_param_path = output_dir / (output_stem + "_params.irpa")
```

Comment:
Hmm... in my suggestion at https://github.com/iree-org/iree/pull/19217#discussion_r1868229488, I had
```python
raise ValueError("If --externalize-params is set, --save-params-to must be set or output must be to a file, not stdout")
```

while here the code defaults to

```python
param_path = Path(args.input_file).parent / "imported_model_params.irpa")
```

if writing to stdout.

I'm struggling to find a use case for writing writing the imported program to stdout but the converted weights to a file, which is why I think it should just be a `ValueError`. I think the implicit/default path handling is a bit tricky to follow here, and writing to the input directory may be unexpected (e.g. it could pollute the git index of a https://github.com/onnx/models checkout). I _do_ think that outputing to stdout for .mlir is reasonable... for simple examples. At the point where you have external parameters in files, I would assume that you would also want the program in a file too.

Thoughts?

Reply:
Ah, yeah. I think that makes more sense. I'll make this change, since I don't see a use case for saving params but not the mlir.